jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/MemoryResource.java

```diff
@@ -0,0 +1,117 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util.resource;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Path;
+import java.util.Objects;
+
+import org.eclipse.jetty.util.IO;
+
+/**
+ * <p>An in memory Resource created from a {@link URL}</p>
+ */
+public class MemoryResource extends Resource
+{
+
+    private final URI _uri;
+    private final long _created = System.currentTimeMillis();
+    private final byte[] _bytes;
+
+    public MemoryResource(URL url)
+    {
+        try
+        {
+            _uri = Objects.requireNonNull(url).toURI();
+            _bytes = IO.readBytes(url.openStream());
+        }
+        catch (IOException | URISyntaxException e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Path getPath()
+    {
+        return null;
```

Comment:
@lorban Well every time I see anybody saying that `getPath` cannot return null, I've said "oh yes it can".   If getPath could always return non-null then we would have succeeded in replacing Resource with Path as there would be a 1:1 mapping.   But there is not.     ResourceCollection is one such resource that cannot always return a Path.

I've been on the lookout for code that makes that kind of assumption (eg getting the Path and using it to get last modified, rather than just asking the resource).   We need to remove such assumptions. We should javadoc that getPath can return null.

Reply:
> > maybe by creating a simple in-memory filesystem?
> 
> heh, uh, this is a BIG undertaking.

Yeah - that's why I want this to be just a single resource - no lists, no directories, no resolve.